lib/pluginUtils.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const assert = require('assert')
+
+function getMeta (fn) {
+  return fn[Symbol.for('plugin-meta')]
+}
+
+function shouldSkipOverride (fn) {
+  return !!fn[Symbol.for('skip-override')]
+}
+
+function checkDependencies (fn) {
+  const meta = getMeta(fn)
+  if (!meta) return
+
+  const deps = meta.dependencies
+  if (!deps) return
+
+  if (deps.fastify) _checkDependencies.call(this, 'Fastify', deps.fastify)
+  if (deps.reply) _checkDependencies.call(this._Reply, 'Reply', deps.reply)
+  if (deps.request) _checkDependencies.call(this._Request, 'Request', deps.request)
```

Comment:
this is an interesting approach, we are actually pattern-matching that fastify, request and reply have been decorated with a given property. I'm 👍 , but what the other thinks?

However, we should change the name `'dependencies'` hints at plugin names (which maybe we should do as well), but this is more `'decorators'`.

Reply:
It does offer flexibility in dependency fulfillment. For example, a plugin may say "requires fastify-foo" because the plugin author expects the `.decorateRequest('foo', 'foo')` from that dependency to be present. But maybe the plugin _user_ knows `fastify-foo2` will satisfy the same requirement. So being able to specify the decorator requirements is definitely an interesting idea. The author of the plugin with the dependency requirement can officially support the plugin he suggested, but the plugin user gets flexibility.

Reply:
The problem is that we cannot check for the plugin "npm name", because it can be present but not attached to our instance.
here I'm checking the decorator, another solution could be add an internal object to Fastify that know which plugins are registered with the current instance and check inside that object.
What do you think?

Reply:
The internal object is what I was thinking.